### PR TITLE
DDF-3298 Resolve SonarQube identified vulnerabilities

### PR DESCRIPTION
**Pull request by josephthweatt**
#### What does this PR do?
Sonarqube has identified potential vulnerabilities in DDF which should be addressed. The issues can be found at https://sonarcloud.io/project/issues?id=ddf&resolved=false&types=VULNERABILITY
#### Who is reviewing it? 
@mcalcote @emmberk 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Choose 2 committers to review/merge the PR. 
@clockard 
@shaundmorris 
#### What are the relevant tickets?
[DDF-3298](https://codice.atlassian.net/browse/)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/ui/search-ui/standard/src/main/webapp/templates/header.layout.handlebars [hunk 1]
```diff
@@ -17,4 +17,4 @@
             {{ui.header}}
         </div>
     </div>
-</div>
\ No newline at end of file
+</div>w
```
Comment (emmberk) [inline catalog/ui/search-ui/standard/src/main/webapp/templates/header.layout.handlebars]: Looks like a typo

File: distribution/sdk/ddf-sso-query-widget/src/main/java/org/codice/ddf/ui/Query.java [hunk 2]
```diff
@@ -95,21 +95,29 @@ public void setFilterBuilder(FilterBuilder filterBuilder) {
...
 
     @Override
-    protected void doGet(HttpServletRequest request, HttpServletResponse response)
-            throws ServletException, IOException {
-        doPost(request, response);
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) {
+        try {
+            doPost(request, response);
+        } catch (ServletException | IOException e) {
+            LOGGER.error("Could not post response.");
```
Comment (emmberk) [inline distribution/sdk/ddf-sso-query-widget/src/main/java/org/codice/ddf/ui/Query.java]: >Error messages should have information that can tell a system administrator how to recover from this condition

https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels#LogLevels-Error

File: distribution/sdk/ddf-sso-query-widget/src/main/java/org/codice/ddf/ui/Query.java [hunk 3]
```diff
@@ -95,21 +95,29 @@ public void setFilterBuilder(FilterBuilder filterBuilder) {
...
+
+        try {
+            PrintWriter writer = response.getWriter();
+            LOGGER.debug("serviceticket request parameter: {}",
+                    request.getParameter(PROXY_TICKET_REQUEST_PARAM));
+            LOGGER.debug("query request parameter: {}", request.getParameter(QUERY_REQUEST_PARAM));
+            String html = createPage(request);
+            writer.println(html);
+        } catch (IOException e) {
+            LOGGER.error("Could not get PrintWriter");
```
Comment (emmberk) [inline distribution/sdk/ddf-sso-query-widget/src/main/java/org/codice/ddf/ui/Query.java]: >Error messages should have information that can tell a system administrator how to recover from this condition

https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels#LogLevels-Error

File: distribution/sdk/ddf-sso-query-widget/src/main/java/org/codice/ddf/ui/QueryForm.java [hunk 4]
```diff
@@ -43,18 +43,26 @@
...
 
     @Override
-    protected void doGet(HttpServletRequest request, HttpServletResponse response)
-            throws ServletException, IOException {
-        doPost(request, response);
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) {
+        try {
+            doPost(request, response);
+        } catch (ServletException | IOException e) {
+            LOGGER.error("Could not post response.");
```
Comment (emmberk) [inline distribution/sdk/ddf-sso-query-widget/src/main/java/org/codice/ddf/ui/QueryForm.java]: >Error messages should have information that can tell a system administrator how to recover from this condition

https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels#LogLevels-Error

File: distribution/sdk/ddf-sso-query-widget/src/main/java/org/codice/ddf/ui/QueryForm.java [hunk 5]
```diff
@@ -43,18 +43,26 @@
...
-        PrintWriter writer = response.getWriter();
-        String html = createPage(request);
-        writer.println(html);
+
+        try {
+            PrintWriter writer = response.getWriter();
+            String html = createPage(request);
+            writer.println(html);
+        } catch (IOException e) {
+            LOGGER.error("Could not get PrintWriter");
```
Comment (emmberk) [inline distribution/sdk/ddf-sso-query-widget/src/main/java/org/codice/ddf/ui/QueryForm.java]: >Error messages should have information that can tell a system administrator how to recover from this condition

https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels#LogLevels-Error

File: distribution/sdk/ddf-sso-widget/pom.xml [hunk 6]
```diff
@@ -39,6 +39,11 @@
             <artifactId>cas-client-core</artifactId>
             <version>3.1.10</version>
         </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>1.7.25</version>
```
Comment (emmberk) [inline distribution/sdk/ddf-sso-widget/pom.xml]: `${org.slf4j.version}`?

File: distribution/sdk/ddf-sso-widget/src/main/java/org/codice/ddf/ui/ExampleSSOWidget.java [hunk 7]
```diff
@@ -28,20 +28,28 @@
...
     protected void doGet(HttpServletRequest request, HttpServletResponse response)
-            throws ServletException, IOException {
+            throws ServletException {
         response.setContentType("text/html");
-        createPage(request, response.getWriter());
+
+        try {
+            createPage(request, response.getWriter());
+        } catch (IOException e) {
+            LOGGER.error("Cannot create page.");
```
Comment (emmberk) [inline distribution/sdk/ddf-sso-widget/src/main/java/org/codice/ddf/ui/ExampleSSOWidget.java]: >Error messages should have information that can tell a system administrator how to recover from this condition

https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels#LogLevels-Error

File: libs/httpproxy/proxy-camel-servlet/src/main/java/org/codice/proxy/http/HttpProxyCamelHttpTransportServlet.java [hunk 8]
```diff
@@ -201,32 +216,36 @@ protected void doService(HttpServletRequest oldRequest, HttpServletResponse resp
...
-                .isTraceEnabled()) {
-            response.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED);
-            return;
+            if ("TRACE".equals(request.getMethod()) && !consumer.getEndpoint()
+                    .isTraceEnabled()) {
+                response.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED);
+                return;
+            }
+        } catch (IOException e) {
+            LOG.error("Could not send error.");
```
Comment (emmberk) [inline libs/httpproxy/proxy-camel-servlet/src/main/java/org/codice/proxy/http/HttpProxyCamelHttpTransportServlet.java]: >Error messages should have information that can tell a system administrator how to recover from this condition

https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels#LogLevels-Error

File: platform/error/platform-error-servlet/src/main/java/org/codice/ddf/platform/error/servlet/ErrorServlet.java [hunk 9]
```diff
@@ -96,10 +109,15 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
...
-                    request,
-                    response);
+
+            try {
+                jettyErrorHandler.handle(request.getRequestURI(),
+                        (org.eclipse.jetty.server.Request) request,
+                        request,
+                        response);
+            } catch (IOException e){
+                LOGGER.error("Problem handling Jetty Error. ", e);
```
Comment (emmberk) [inline platform/error/platform-error-servlet/src/main/java/org/codice/ddf/platform/error/servlet/ErrorServlet.java]: >Error messages should have information that can tell a system administrator how to recover from this condition

https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels#LogLevels-Error

**Comment by coyotesqrl** [comment]
Because of the reach of the changes cause by the update to `PersistentStore`, I think it should be in a separate PR from the other fixes.

File: platform/persistence/platform-persistence-commands/src/main/java/org/codice/ddf/persistence/commands/AbstractStoreCommand.java [hunk 10]
```diff
@@ -49,11 +50,11 @@ public Object execute() {
 
         try {
 
-            if (PersistentStore.PERSISTENCE_TYPES.contains(type)) {
+            if (PERSISTENCE_TYPES.hasType(type)) {
                 storeCommand();
             } else {
                 console.println("Type passed in was not correct. Must be one of "
-                        + PersistentStore.PERSISTENCE_TYPES + ".");
+                        + PERSISTENCE_TYPES.values() + ".");
```
Comment (emmberk) [inline platform/persistence/platform-persistence-commands/src/main/java/org/codice/ddf/persistence/commands/AbstractStoreCommand.java]: How does `PERSISTENCE_TYPES.values()`  print out?

File: distribution/sdk/ddf-sso-query-widget/src/main/java/org/codice/ddf/ui/Query.java (hunk 2 above)
Comment (josephthweatt) [reply distribution/sdk/ddf-sso-query-widget/src/main/java/org/codice/ddf/ui/Query.java]: Alright, so just put all of these back?

File: platform/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java [hunk 11]
```diff
@@ -97,7 +99,7 @@ public void startProxy() throws Exception {
                         e);
                 host = "localhost";
             }
-            endpointName = ((HttpProxyServiceImpl) httpProxyService).start("0.0.0.0:" + httpPort,
```
Comment (vinamartin) [inline platform/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java]: ❓ Do you still need to have that colon bit on the end of the constant?

File: platform/security/servlet/security-servlet-logout/src/main/java/org/codice/ddf/security/servlet/logout/LocalLogoutServlet.java [hunk 12]
```diff
@@ -96,15 +96,19 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
                 }
             }
             redirectUrlBuilder.addParameters(params);
+
             response.sendRedirect(redirectUrlBuilder.build()
                     .toString());
         } catch (URISyntaxException e) {
             LOGGER.debug("Invalid URI", e);
+        } catch (IOException e) {
+            LOGGER.error("Send Redirect failed. ", e);
```
Comment (mcalcote) [inline platform/security/servlet/security-servlet-logout/src/main/java/org/codice/ddf/security/servlet/logout/LocalLogoutServlet.java]: ❓ Do you still need the `throws IOException` on the method with this catch?

File: platform/solr/platform-solr-server-standalone/src/main/webapp/js/angular/controllers/collections.js [hunk 13]
```diff
@@ -52,7 +52,6 @@ solrAdminApp.controller('CollectionsController',
                   }
               }
               if ($routeParams.collection && !$scope.collection) {
-                  alert("No collection called " + $routeParams.collection)
```
Comment (mcalcote) [inline platform/solr/platform-solr-server-standalone/src/main/webapp/js/angular/controllers/collections.js]: ❓ Is there another mechanism in place to eliminate the need to alert the user using this approach?

Comment (josephthweatt) [reply platform/solr/platform-solr-server-standalone/src/main/webapp/js/angular/controllers/collections.js]: Not sure. If you feel it is important that the user knows that the collection does not exist, then I could put this line back

Comment (clockard) [reply platform/solr/platform-solr-server-standalone/src/main/webapp/js/angular/controllers/collections.js]: This file is copied from solr I believe so we shouldn't be changing it.

Comment (clockard) [reply platform/solr/platform-solr-server-standalone/src/main/webapp/js/angular/controllers/collections.js]: That goes for everything in platform-solr-server-standalone

**Comment by josephthweatt** [comment]
build now

**Comment by josephthweatt** [comment]
build now

**Comment by josephthweatt** [comment]
build now

File: libs/httpproxy/proxy-camel-servlet/src/main/java/org/codice/proxy/http/HttpProxyCamelHttpTransportServlet.java [hunk 14]
```diff
@@ -194,28 +207,36 @@ protected void doService(HttpServletRequest oldRequest, HttpServletResponse resp
...
-      return;
+      if ("TRACE".equals(request.getMethod()) && !consumer.getEndpoint().isTraceEnabled()) {
+        response.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED);
+        return;
+      }
+    } catch (IOException e) {
+      LOG.warn("Could not send error due to: ", e);
+    } finally {
+      if (consumer == null) {
+        return;
```
[the remark this answers is not in the document]
Comment (josephthweatt) [reply libs/httpproxy/proxy-camel-servlet/src/main/java/org/codice/proxy/http/HttpProxyCamelHttpTransportServlet.java]: I'll just make that `if` independent of the block

**Comment by josephthweatt** [comment]
build now

**Comment by josephthweatt** [comment]
build now

File: platform/error/platform-error-servlet/src/main/java/org/codice/ddf/platform/error/servlet/ErrorServlet.java [hunk 15]
```diff
@@ -84,13 +87,24 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
...
 
     if (errorHandler != null) {
-      errorHandler.handleError(
-          Integer.parseInt(code), message, type, throwable, uri, request, response);
+      int codeInt;
+
+      try {
+        codeInt = Integer.parseInt(code);
+      } catch (NumberFormatException e) {
+        codeInt = 0;
```
Comment (stustison) [inline platform/error/platform-error-servlet/src/main/java/org/codice/ddf/platform/error/servlet/ErrorServlet.java]: Can we set this to 500 instead of 0?

Comment (josephthweatt) [reply platform/error/platform-error-servlet/src/main/java/org/codice/ddf/platform/error/servlet/ErrorServlet.java]: sure, just a sec

File: libs/httpproxy/proxy-camel-servlet/src/main/java/org/codice/proxy/http/HttpProxyCamelHttpTransportServlet.java [hunk 16]
```diff
@@ -38,6 +38,19 @@
  * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
  * express or implied. See the License for the specific language governing permissions and
  * limitations under the License.
+ *
+ * <p>Licensed to the Apache Software Foundation (ASF) under one or more contributor license
```
Comment (clockard) [inline libs/httpproxy/proxy-camel-servlet/src/main/java/org/codice/proxy/http/HttpProxyCamelHttpTransportServlet.java]: Why was this added? Looks like we have it duplicated already. Should we be removing one instead of adding one.

Comment (josephthweatt) [reply libs/httpproxy/proxy-camel-servlet/src/main/java/org/codice/proxy/http/HttpProxyCamelHttpTransportServlet.java]: @clockard This must have gotten duplicated when I did the formatter, I'll remove the duplicates

**Comment by josephthweatt** [comment]
build now

**Comment by josephthweatt** [comment]
build now

File: libs/httpproxy/proxy-camel-servlet/src/main/java/org/codice/proxy/http/HttpProxyCamelHttpTransportServlet.java [hunk 17]
```diff
@@ -194,28 +166,36 @@ protected void doService(HttpServletRequest oldRequest, HttpServletResponse resp
...
-      response.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED);
-      return;
+      if ("TRACE".equals(request.getMethod()) && !consumer.getEndpoint().isTraceEnabled()) {
+        response.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED);
+        return;
+      }
+    } catch (IOException e) {
+      LOG.warn("Could not send error due to: ", e);
+    } finally {
+      if (consumer == null) {
```
Comment (paouelle) [inline libs/httpproxy/proxy-camel-servlet/src/main/java/org/codice/proxy/http/HttpProxyCamelHttpTransportServlet.java]: ❗️ Never a good idea to suddenly change the way a try/catch block like this exits in the finally. If the blocks ends up exiting because of runtime exceptions, for example, then those are swallowed. Same for errors ...

Comment (josephthweatt) [reply libs/httpproxy/proxy-camel-servlet/src/main/java/org/codice/proxy/http/HttpProxyCamelHttpTransportServlet.java]: Oddly enough I though I took this out. Thanks for catching it, I'll change it when I get in tomorrow

**Comment by josephthweatt** [comment]
build now

**Comment by josephthweatt** [comment]
build now

**Comment by LinkMJB** [comment]
build now

**Comment by LinkMJB** [comment]
build now

**Comment by clockard** [comment]
Looks like all comments have been addressed. Merging.

**Comment by clockard** [comment]
@josephthweatt please make a port pr for the 2.11.x branch

**Comment by rzwiefel** [comment]
@clockard @josephthweatt I went ahead and made a PR here https://github.com/codice/ddf/pull/2418
